pyproject.toml

```diff
@@ -49,6 +49,12 @@ codecov = ">=2.1.12"
 pytest-cov = "^4.1.0"
 pre-commit = ">=3.3.2"
 
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
```

Comment:
Would it make sense to port other docs dependencies to this group? I'm not sure of how this would impact the builds, but it might make for deduplicated dependency specification code.

Reply:
I like this idea to centralize the dependencies. We would probably make a tweak to the other build pipelines to change `poetry install --all-extras` to `poetry install --all-extras --without docs`

Reply:
is this something to include in this PR? If so, can either of you tell me what needs to be included here? what other docs? Thanks!

Reply:
@kenibrewer great thinking! @gwaybio I think this would involve porting the requirements found in `docs/requirements.txt` to the `pyproject.toml` by way of `poetry add <pkg name> --group docs` or manually adding them to the `pyproject.toml` file and issuing a `poetry lock --no-update` afterwards. If this is implemented I believe the `docs/requirements.txt` file and lines 20-21 from `.readthedocs.yml` could be removed  (they'd become duplicative).

Reply:
The other required changes would be:
- `.devcontainer/postCreateCommand.sh`:
  Replace line 4 with: `poetry install --all-extras --without docs`
- `.github/workflows/pytest.yml`
  Replace line 58 with: `run: poetry install --no-interaction --no-root  --all-extras --without docs`

Reply:
I tried but got a nifty error:

```bash
vscode ➜ /workspaces/pycytominer (nb-syntax-fix) $ poetry add sphinx --group docs
Using version ^7.2.5 for sphinx

Updating dependencies
Resolving dependencies... (0.0s)

The current project's Python requirement (>=3.8) is not compatible with some of the required packages Python requirement:
  - sphinx requires Python >=3.9, so it will not be satisfied for Python >=3.8,<3.9

Because no versions of sphinx match >7.2.5,<8.0.0
 and sphinx (7.2.5) requires Python >=3.9, sphinx is forbidden.
So, because pycytominer depends on sphinx (^7.2.5), version solving failed.
```

I don't think it's worth changing our package environment to satisfy docs requirements, especially given that they work fine at the moment. My vote is to keep docs requirements separate

Reply:
I added @kenibrewer 's changes in b223f3edd5e01732a44943f651ad796cc67a7062

Reply:
> ```shell
> Because no versions of sphinx match >7.2.5,<8.0.0
>  and sphinx (7.2.5) requires Python >=3.9, sphinx is forbidden.
> So, because pycytominer depends on sphinx (^7.2.5), version solving failed.
> ```

In the [Sphinx Changelog](https://www.sphinx-doc.org/en/master/changes.html), it looks like they dropped support for Python 3.8 in version 7.2. Can you try specifying the previous version using `poetry add sphinx@~7.1 --group docs` ?

It looks like my suggested changes to he poetry install command resulted in pytest not being available for the pytest workflow. Can you try replacing `--all-extras --without docs` with `--with cell_locations,collate,dev`?

Reply:
Looks good!